backend_new/src/controllers/auth.controller.ts

```diff
@@ -0,0 +1,111 @@
+import {
+  Controller,
+  Get,
+  Request,
+  Response,
+  Post,
+  UsePipes,
+  ValidationPipe,
+  Body,
+  BadRequestException,
+  Put,
+  UseGuards,
+} from '@nestjs/common';
+import { ApiBody, ApiOkResponse, ApiOperation, ApiTags } from '@nestjs/swagger';
+import { Response as ExpressResponse } from 'express';
+import { defaultValidationPipeOptions } from '../utilities/default-validation-pipe-options';
+import { SuccessDTO } from '../dtos/shared/success.dto';
+import { AuthService, REFRESH_COOKIE_NAME } from '../services/auth.service';
+import { RequestMfaCode } from '../dtos/mfa/request-mfa-code.dto';
+import { RequestMfaCodeResponse } from '../dtos/mfa/request-mfa-code-response.dto';
+import { Confirm } from '../dtos/auth/confirm.dto';
+import { UpdatePassword } from '../dtos/auth/update-password.dto';
+import { MfaAuthGuard } from '../guards/mfa.guard';
+import { JwtAuthGuard } from '../guards/jwt.guard';
+import { OptionalAuthGuard } from '../guards/optional.guard';
+import { Login } from '../dtos/auth/login.dto';
+import { mapTo } from '../utilities/mapTo';
+import { User } from '../dtos/users/user.dto';
+
+@Controller('auth')
+@ApiTags('auth')
+@UsePipes(new ValidationPipe(defaultValidationPipeOptions))
+export class AuthController {
+  constructor(private readonly authService: AuthService) {}
+
+  @Post('login')
+  @ApiOperation({ summary: 'Login', operationId: 'login' })
+  @ApiOkResponse({ type: SuccessDTO })
+  @ApiBody({ type: Login })
+  @UseGuards(MfaAuthGuard)
+  async login(
+    @Request() req,
+    @Response({ passthrough: true }) res: ExpressResponse,
+  ): Promise<SuccessDTO> {
+    return await this.authService.setCredentials(res, mapTo(User, req['user']));
+  }
+
+  @Get('logout')
+  @ApiOperation({ summary: 'Logout', operationId: 'logout' })
+  @ApiOkResponse({ type: SuccessDTO })
+  @UseGuards(JwtAuthGuard)
+  async logout(
+    @Request() req,
+    @Response({ passthrough: true }) res,
+  ): Promise<SuccessDTO> {
+    return await this.authService.clearCredentials(
+      res,
+      mapTo(User, req['user']),
+    );
+  }
+
+  @Post('request-mfa-code')
+  @ApiOperation({ summary: 'Request mfa code', operationId: 'requestMfaCode' })
+  @ApiOkResponse({ type: RequestMfaCodeResponse })
+  async requestMfaCode(
+    @Body() dto: RequestMfaCode,
+  ): Promise<RequestMfaCodeResponse> {
+    return await this.authService.requestMfaCode(dto);
+  }
+
+  @Get('requestNewToken')
```

Comment:
so if you do the auth/login endpoint while on http://localhost:3101/api that sets the token for you on the site (since thats more or less a UI for the backend)

then you can just hit the new token endpoint through there

Reply:
👍